### PR TITLE
migrate to central package management

### LINKED ISSUE #299: Consider using Directory.Packages.props

### PR DESCRIPTION
**Pull request by oising**
all tests and src now using Directory.Packages.props. Also updated targets (for sourcelink)

closes #299

### CODE REVIEW

**Comment by oising** [comment]
packages were consolidated using the highest version used in the solution, if differences were encountered (I wrote a pwsh script some time ago to do the grunt work.)

File: Directory.Packages.props [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+ 		<PackageVersion Include="MQTTnet" Version="4.3.6.1152" />
+ 		<PackageVersion Include="Newtonsoft.Json" Version="13.0.3" />
+ 		<PackageVersion Include="Nullable" Version="1.3.1" />
+ 		<PackageVersion Include="System.Memory" Version="4.5.5" />
+ 		<PackageVersion Include="System.Text.Encodings.Web" Version="8.0.0" />
+ 		<PackageVersion Include="System.Text.Json" Version="8.0.4" />
+ 		<PackageVersion Include="xunit" Version="2.9.0" />
+ 		<PackageVersion Include="xunit.runner.console" Version="2.9.0" />
+ 		<PackageVersion Include="xunit.runner.visualstudio" Version="2.8.2" />
+
```
Comment (jskeet) [inline Directory.Packages.props:23]: Tiny nit: remove blank line. (I'm happy to do it in a follow-up PR if you want.)

**Review by jskeet — APPROVED** [review]
Nice, thanks!

File: Directory.Packages.props (hunk 1 above)
Comment (oising) [reply Directory.Packages.props:23]: Side effect of the script. I'll clean it up.

**Comment by jskeet** [comment]
Thanks, will squash and merge on green.

**Pull request merged**